core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PartitionImpl.java

```diff
@@ -140,8 +140,8 @@ public void setPool(Pool pool) {
   public boolean equals(Object obj) {
     if (obj == null) return false;
     if (obj == this) return true;
-    if (!(obj instanceof PartitionImpl)) return false;
-    PartitionImpl them = (PartitionImpl) obj;
+    if (!(obj instanceof Partition)) return false;
```

Comment:
Yes, but one might be a proxy. So, either it has to be deproxied (and the models know about Hibernate), or they should just check using the interface.